chalice/cli/reload.py

```diff
@@ -0,0 +1,71 @@
+import os
+import subprocess
+import sys
+import threading
+import time
+import types
+
+import six
+from typing import Optional, Dict, Type  # noqa
+
+from chalice.constants import AUTORELOAD_INTERVAL
+
+
+class Reloader(threading.Thread):
+    def __init__(self, autoreload=True):
+        # type: (bool) -> None
+        super(Reloader, self).__init__()
+        self.autoreload = autoreload
+        self.triggered = False
+        self.mtimes = {}  # type: Dict[str, float]
+
+    def __enter__(self):
+        # type: () -> Reloader
+        if self.autoreload:
+            self.setDaemon(True)
+            self.start()
+        return self
+
+    def __exit__(self,
+                 exc_type,  # type: Optional[Type[BaseException]]
+                 exc_value,  # type: Optional[BaseException]
+                 traceback  # type: Optional[types.TracebackType]
+                 ):
+        # type: (...) -> None
+        if exc_type is KeyboardInterrupt and self.triggered:
+            sys.exit(1)
+
+    def run(self):
+        # type: () -> None
+        while True:
+            time.sleep(AUTORELOAD_INTERVAL)
+            if self.find_changes():
+                self.reload()
+
+    def find_changes(self):
+        # type: () -> bool
+        for module in list(sys.modules.values()):
```

Comment:
* No need to cast to `list` here. 
* Can we scope this to just the files that are in their Chalice project?

Reply:
> No need to cast to ``list`` here.

Why?

>  Can we scope this to just the files that are in their Chalice project?

We can.. but I think it's better to check all modules.